google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/util/AppEngineUtil.java

```diff
@@ -61,24 +63,23 @@ private AppEngineUtil() {
    * @return a list of {@link AppEngineArtifactDeploymentSource}'s
    */
   public static List<AppEngineArtifactDeploymentSource> createArtifactDeploymentSources(
-      @NotNull Project project) {
+      @NotNull final Project project) {
     List<AppEngineArtifactDeploymentSource> sources = Lists.newArrayList();
-    AppEngineProjectService aeProjectHelper = AppEngineProjectService.getInstance();
+    AppEngineProjectService projectService = AppEngineProjectService.getInstance();
+    AppEngineAssetProvider assetProvider = AppEngineAssetProvider.getInstance();
 
     for (Module module : ModuleManager.getInstance(project).getModules()) {
-      Collection<Artifact> artifacts = ArtifactUtil.getArtifactsContainingModuleOutput(module);
+      XmlFile appEngineWebXml = assetProvider.loadAppEngineStandardWebXml(project, module);
+      final AppEngineEnvironment environment
+          = projectService.getAppEngineArtifactEnvironment(appEngineWebXml);
+
+      boolean isFlexCompat = projectService.isFlexCompat(appEngineWebXml);
 
+      Collection<Artifact> artifacts = ArtifactUtil.getArtifactsContainingModuleOutput(module);
       for (Artifact artifact : artifacts) {
-        boolean hasStandardArtifacts
-            = aeProjectHelper.containsAppEngineStandardArtifacts(project, artifacts);
-        boolean addFlexArtifact
-            = !hasStandardArtifacts || aeProjectHelper.isFlexCompat(project, artifact);
-        AppEngineEnvironment environment
-            = aeProjectHelper.getAppEngineArtifactEnvironment(project, artifact);
-
-        if (environment != null
-            && (environment.isStandard()
-            || (environment.isFlexible() && addFlexArtifact))) {
+        boolean isStandardProject = environment.isStandard() || isFlexCompat;
```

Comment:
yes. a couple things to rename here I noticed too. getAppEngineArtifactEnvironment -> getAppEngineEnvironment. Also I can move that boolean up out of the inner loop.